evennia/contrib/turnbattle.py

```diff
@@ -0,0 +1,520 @@
+            fighter.db.Combat_ActionsLeft = 1 #Actions remaining - start of turn adds to this, turn ends when it reaches 0
+            fighter.db.Combat_TurnHandler = self #Add a reference to this script to the character
+            fighter.db.Combat_LastAction = "null" #Track last action taken in combat
+        # Roll initiative and sort the list of fighters depending on who rolls highest to determine turn order.
+        # The initiative roll is determined by the roll_init function and can be customized easily.
+        ordered_by_roll = sorted(self.db.fighters, key=roll_init, reverse=True)
+        self.db.fighters = ordered_by_roll
+        # Announce the turn order.
+        self.obj.msg_contents("Turn order is: %s " % ", ".join(obj.key for obj in self.db.fighters))
+        "Set up the current turn and turn timeout delay."
+        self.db.turn = 0
+        self.db.timer = 30 # 30 seconds
+        
+    def at_stop(self):
+        """
+        Called at script termination.
+        """
+        for fighter in self.db.fighters:
+            combat_cleanup(fighter) #Clean up the combat attributes for every fighter.
+    
+    def at_repeat(self):
+        """
+        Called once every self.interval seconds.
+        """
+        currentchar = self.db.fighters[self.db.turn] # Note the current character in the turn order.
+        self.db.timer -= 1 # Count down the timer by one second.
+        
+        # If the current character has no actions remaining, go to the next turn.
+        if not currentchar.db.Combat_ActionsLeft:
+            self.next_turn()
+            return
+            
+        # Warn the current character if they're about to time out.
+        if self.db.timer == 10: # 10 seconds left
+            currentchar.msg("WARNING: About to time out!")
+        
+        # Force current character to disengage if timer runs out.
+        if self.db.timer <= 0:
+            currentchar.db.Combat_LastAction = "disengage" # Set last action to 'disengage'
+            currentchar.db.Combat_ActionsLeft = 0 # Set actions remaining to 0
+            self.obj.msg_contents("%s's turn timed out!" % currentchar)
+            self.next_turn()
+            
+    def next_turn(self):
+        """
+        Advances to the next character in the turn order.
+        """
+        
+        # Check to see if every character disengaged as their last action. If so, end combat.
+        DisengageCheck = True
+        for fighter in self.db.fighters:
+            if fighter.db.Combat_LastAction != "disengage": # If a character has done anything but disengage
+                DisengageCheck = False
+        if DisengageCheck == True: # All characters have disengaged
+            self.obj.msg_contents("All fighters have disengaged! Combat is over!")
+            self.stop() # Stop this script and end combat.
+            return
+        
+        # Check to see if only one character is left standing. If so, end combat.
+        DefeatedCharacters = 0
```

Comment:
Python PEP8 standard (and thus Evennia's style guide) dictates that local variables are always written in lowercase with underscores. CamelCase is only ever used for class names.